<a href="https://colab.research.google.com/github/shaiza1iman/loan-approval-prediction/blob/main/notebooks/loan_approval_prediction.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# Loan Approval Prediction using Logistic Regression

## Phase 1 — Problem Understanding

### Problem Statement

The goal of this project is to build a machine learning model that predicts whether a loan application is likely to be approved or rejected based on applicant information.

### Machine Learning Problem

This is a supervised binary classification problem.

- Target: Loan approval status
- Classes: Approved / Rejected
- Model: Logistic Regression

### Real-World Application

A loan approval prediction system can be used as an educational example of how machine learning can classify loan applications based on applicant characteristics.

> Disclaimer: This project is an educational machine learning demonstration and is not a real-world banking decision system.

## Target and Input Features

### Target Variable

The target variable is the **loan approval status**. The model will predict whether a loan application is approved or rejected.

### Input Features

Input features are the applicant and loan-related information used by the model to make its prediction.

The exact input features will be identified after we inspect the dataset in Phase 2.

## Why Classification?

This project is a classification problem because the model needs to predict one of two possible outcomes:

- Approved
- Rejected

Logistic Regression is appropriate for this project because it is a classification algorithm that can predict the probability of belonging to a class and then make a classification decision.

In [ ]:
import pandas as pd

df = pd.read_csv("Loan_Data.csv")

df.head()

,Loan_ID,Gender,Married,Dependents,Education,Self_Employed,ApplicantIncome,CoapplicantIncome,LoanAmount,Loan_Amount_Term,Credit_History,Property_Area,Loan_Status
0,LP001002,Male,No,0,Graduate,No,5849,0.0,NaN,360.0,1.0,Urban,Y
1,LP001003,Male,Yes,1,Graduate,No,4583,1508.0,128.0,360.0,1.0,Rural,N
2,LP001005,Male,Yes,0,Graduate,Yes,3000,0.0,66.0,360.0,1.0,Urban,Y
3,LP001006,Male,Yes,0,Not Graduate,No,2583,2358.0,120.0,360.0,1.0,Urban,Y
4,LP001008,Male,No,0,Graduate,No,6000,0.0,141.0,360.0,1.0,Urban,Y


In [ ]:
df.shape

(614, 13)

In [ ]:
df.info()


<class 'pandas.core.frame.DataFrame'>
RangeIndex: 614 entries, 0 to 613
Data columns (total 13 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Loan_ID            614 non-null    object 
 1   Gender             601 non-null    object 
 2   Married            611 non-null    object 
 3   Dependents         599 non-null    object 
 4   Education          614 non-null    object 
 5   Self_Employed      582 non-null    object 
 6   ApplicantIncome    614 non-null    int64  
 7   CoapplicantIncome  614 non-null    float64
 8   LoanAmount         592 non-null    float64
 9   Loan_Amount_Term   600 non-null    float64
 10  Credit_History     564 non-null    float64
 11  Property_Area      614 non-null    object 
 12  Loan_Status        614 non-null    object 
dtypes: float64(4), int64(1), object(8)
memory usage: 62.5+ KB


In [ ]:
df["Loan_Status"].value_counts()
df.columns.tolist()


['Loan_ID',
 'Gender',
 'Married',
 'Dependents',
 'Education',
 'Self_Employed',
 'ApplicantIncome',
 'CoapplicantIncome',
 'LoanAmount',
 'Loan_Amount_Term',
 'Credit_History',
 'Property_Area',
 'Loan_Status']

## Dataset Information

### Dataset Source

Dataset: Eligibility Prediction for Loan

Source: Kaggle — Devzohaib

The dataset contains information about loan applicants and their loan applications. The target variable is `Loan_Status`, which indicates whether the loan was approved or rejected.

### Dataset Dimensions

- Rows: 614
- Columns: 13

### Column Overview

| Column | Description |
|---|---|
| `Loan_ID` | Unique identifier for the loan application |
| `Gender` | Gender of the applicant |
| `Married` | Marital status of the applicant |
| `Dependents` | Number of dependents |
| `Education` | Education level |
| `Self_Employed` | Whether the applicant is self-employed |
| `ApplicantIncome` | Income of the applicant |
| `CoapplicantIncome` | Income of the co-applicant |
| `LoanAmount` | Requested loan amount |
| `Loan_Amount_Term` | Loan repayment term |
| `Credit_History` | Applicant's credit history information |
| `Property_Area` | Location type of the property |
| `Loan_Status` | Target variable: loan approval status |

### Target Distribution

- Approved (`Y`): 422
- Rejected (`N`): 192